spark/src/main/scala/geotrellis/spark/io/AttributeStore.scala

```diff
@@ -9,6 +9,13 @@ trait AttributeStore[F[_]] extends AttributeCaching[F] {
   def readAll[T: Format](attributeName: String): Map[LayerId, T]
   def write[T: Format](layerId: LayerId, attributeName: String, value: T): Unit
   def layerExists(layerId: LayerId): Boolean
+  def delete(layerId: Option[LayerId], attributeName: Option[String]): Unit
+  def delete(layerId: LayerId, attributeName: String): Unit =
+    delete(Some(layerId), Some(attributeName))
+  def delete(layerId: LayerId): Unit =
+    delete(Some(layerId), None)
+  def delete(attributeName: String): Unit =
```

Comment:
That makes sense. Does it throw if the layer/attribute does not exist or silent pass through?